examples/playground/index.tsx

```diff
@@ -0,0 +1,134 @@
+import React, { useState, useEffect } from 'react';
+import ReactDOM from 'react-dom';
+import ReactDOMServer from 'react-dom/server';
+
+import { useLazyPhase, MODE } from 'react-loosely-lazy';
+import { listeners } from './utils';
+
+import {
+  ForPaintComponents,
+  AfterPaintComponents,
+  LazyComponents,
+  CustomWaitComponents,
+  buildServerComponents,
+  buildClientComponents,
+} from './components';
+
+const steps = [
+  'SSR',
+  'PAINT LOADING',
+  'PAINT FETCHING',
+  'PAINT READY',
+  'AFTER LOADING',
+  'AFTER FETCHING',
+  'AFTER READY', // also 'LAZY READY'
+  'CUSTOM LOADING',
+  'CUSTOM FETCHING',
+  'CUSTOM READY',
+];
+
+/**
+ * Controls App
+ */
+const Controls = () => {
+  const [stepIndex, setStepIndex] = useState(0);
+  const [step, setStep] = useState(steps[stepIndex]);
+  const resetSteps = () => setStepIndex(0);
+
+  useEffect(() => {
+    (window as any).step = steps[stepIndex];
+    setStep(steps[stepIndex]);
+  }, [stepIndex, setStep]);
+
+  useEffect(() => {
+    listeners.forEach(l => l(step));
+  }, [step]);
+
+  return (
+    <div>
+      <h1>{window.location.hash || 'hydration'} example</h1>
+      <p>
+        <a href="#render" onClick={() => resetSteps()}>
+          Switch to render
+        </a>
+        {' | '}
+        <a href="#hydration" onClick={() => resetSteps()}>
+          Switch to hydration
+        </a>
+        {' | '}
+        <a href="#failssr" onClick={() => resetSteps()}>
+          Switch to failing ssr
+        </a>
+      </p>
+      <ul>
+        {steps.map(v => (
+          <li key={v}>
+            <label>
+              <input type="radio" checked={step === v} readOnly /> {v}
+            </label>
+          </li>
+        ))}
+      </ul>
+      <button onClick={() => setStepIndex(stepIndex + 1)}>Next step</button>
```

Comment:
Is it possible for this to loop back to step 1 and use mod? Or at least cap the next step, because clicking it on the last step blows up the app.